### PR TITLE
Issue where filtering by date if there is a module that adds extra columns and has the created_at

### PR DESCRIPTION
**Pull request by alucardatem**
### Description (*)
Issue where filtering by date if there is a module that adds extra columns and has the created_at as well, it ends up with error on filtering

error: 
```main.CRITICAL: SQLSTATE[23000]: Integrity constraint violation: 1052 Column 'created_at' in where clause is ambiguous, query was: SELECT COUNT(*) FROM `sales_order_grid` AS `main_table`
 LEFT JOIN `braintree_transaction_details` ON braintree_transaction_details.order_id = main_table.entity_id
 RIGHT JOIN `sales_order` ON main_table.entity_id = sales_order.entity_id WHERE  (((`created_at` >= '2022-11-01 00:00:00'))) AND (((`created_at` <= '2022-11-04 23:59:59')))```

change was proposed also in the security patches - in MDVA-38346 on git

### Fixed Issues (if relevant)

1. Fixes magento/magento2#<issue_number>

### Manual testing scenarios (*)

1. ...
2. ...

### Contribution checklist (*)
 - [x] Pull request has a meaningful description of its purpose
 - [x] All commits are accompanied by meaningful commit messages
 - [x] All new or changed code is covered with unit/integration tests (if applicable)
 - [x] README.md files for modified modules are updated and included in the pull request if any [README.md predefined sections](https://github.com/magento/devdocs/wiki/Magento-module-README.md) require an update
 - [x] All automated tests passed successfully (all builds are green)

### CODE REVIEW

File: app/code/Magento/Sales/Plugin/Model/ResourceModel/Order/OrderGridCollectionFilter.php [hunk 1]
```diff
@@ -55,7 +55,7 @@ public function aroundAddFieldToFilter(
                 }
             }
 
-            $fieldName = $subject->getConnection()->quoteIdentifier($field);
+            $fieldName = "main_table.".$subject->getConnection()->quoteIdentifier($field);
```
Comment (Den4ik) [inline app/code/Magento/Sales/Plugin/Model/ResourceModel/Order/OrderGridCollectionFilter.php]: Could you please format code with spaces
```suggestion
            $fieldName = "main_table." . $subject->getConnection()->quoteIdentifier($field);
```

**Review by Den4ik — CHANGES_REQUESTED** [review]

**Comment by Den4ik** [comment]
@magento run all tests

**Review by Den4ik — CHANGES_REQUESTED** [review]
@alucardatem Thanks for your work on this PR.
Could you please fix failed Integrations tests

**Comment by alucardatem** [comment]
Sure thing. Will do.

**Comment by engcom-Echo** [comment]
@magento run all tests

**Comment by alucardatem** [comment]
@magento run all tests

**Comment by alucardatem** [comment]
#magento run Database Compare, Functional Tests CE, Integration Tests, Magento Health Index, Sample Data Tests CE, 
Semantic Version Checker, Static Tests, Unit Tests

**Comment by khoimm92** [comment]
This issue persists on the latest version.

**Comment by alucardatem** [comment]
@khoimm92 beyter apply that as a patch as since tests not running or i do not know how to make them run properly this will never get fixed

**Pull request closed without merging**